BASE/src/ServerTelemetryChannel/Implementation/TransmissionPolicy/AuthenticationTransmissionPolicy.cs

```diff
@@ -0,0 +1,120 @@
+            {
+                throw new ArgumentNullException(nameof(transmitter));
+            }
+
+            this.backoffLogicManager = transmitter.BackoffLogicManager;
+
+            base.Initialize(transmitter);
+            transmitter.TransmissionSent += this.HandleTransmissionSentEvent;
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// This method subscribes to the <see cref="Transmitter.TransmissionSent"/> event.
+        /// This encapsulates all <see cref="HttpWebResponseWrapper.StatusCode"/> related to Authentication (AAD) scenarios.
+        /// </summary>
+        /// <remarks>
+        /// AN EXPLANATION OF THE STATUS CODES:
+        /// - <see cref="ResponseStatusCodes.BadRequest"/>
+        /// "HTTP/1.1 400 Incorrect API was used - v2 API does not support authentication".
+        /// This indicates that the AI resource was configured for AAD, but SDK was not.
+        /// This is a configuration issue and is not recoverable from the client side. 
+        /// If the customer chooses to disable AAD in the AI Resource, the SDK could resume sending.
+        /// - <see cref="ResponseStatusCodes.Unauthorized"/>
+        /// "HTTP/1.1 401 Unauthorized - please provide the valid authorization token".
+        /// This indicates that the authorization token was either absent, invalid, or expired.
+        /// The root cause is not known and we should throttle retries.
+        /// - <see cref="ResponseStatusCodes.Forbidden"/>
+        /// "HTTP/1.1 403 Forbidden - provided credentials do not grant the access to ingest the telemetry into the component".
+        /// This indicates the configured identity does not have permissions to publish to this resource.
+        /// This is a configuration issue and is not recoverable from the client side. 
+        /// This can be recovered if the user changes the AI Resource's configured Access Control.
+        /// </remarks>
+        private void HandleTransmissionSentEvent(object sender, TransmissionProcessedEventArgs e)
+        {
+            if (this.Enabled && e.Response != null)
+            {
+                switch (e.Response.StatusCode)
+                {
+                    case ResponseStatusCodes.BadRequest:
+                    case ResponseStatusCodes.Unauthorized:
+                    case ResponseStatusCodes.Forbidden:
+                        this.ApplyThrottlePolicy(e);
+                        break;
+                }
+            }
+        }
+
+        private void ApplyThrottlePolicy(TransmissionProcessedEventArgs e)
+        {
+            this.MaxSenderCapacity = 0;
+            this.MaxBufferCapacity = null;
+            this.MaxStorageCapacity = null;
+
+            this.LogCapacityChanged();
+            this.Apply();
+
+            this.backoffLogicManager.ReportBackoffEnabled(e.Response.StatusCode);
```

Comment:
it doesn't look like the backofflogicmanager is used anywhere but in this line.. which seems incorrect.

Reply:
We have multiple threads now about BackoffLogicManager... , so i'm sorry if i've lost the current thread. This may be easier to discuss on a call.

I based my implementation off of `ThrottlingTransmissionPolicy`
https://github.com/microsoft/ApplicationInsights-dotnet/blob/ce50b28c693fbb88dd749065039d9c1c57443ac9/BASE/src/ServerTelemetryChannel/Implementation/TransmissionPolicy/ThrottlingTransmissionPolicy.cs#L62-L72

At line 65, `ThrottlingTransmissionPolicy` calls `this.backoffLogicManager.GetBackOffTimeInterval` and provides the value from the retry header.
This triggers the backofflogicmanager to begin the exponential backoff. 
I tried to discribe this above.

I can probably fake this by providing a custom formatted value. But I'll need to do more testing around this specifically.

Reply:
What's strange to me is that in the entire SDK, we only have one implementation of TransmissionPolicy that is correctly resetting the backoff:

`PartialSuccessTransmissionPolicy` is the only class that calls `this.backoffLogicManager.ResetConsecutiveErrors();`
https://github.com/microsoft/ApplicationInsights-dotnet/blob/ce50b28c693fbb88dd749065039d9c1c57443ac9/BASE/src/ServerTelemetryChannel/Implementation/TransmissionPolicy/PartialSuccessTransmissionPolicy.cs#L105-L112

I don't have an explanation for this and would need more time to fully investigate